From: Kaixu Xia <kaixuxia@tencent.com>
Subject: mm/damon/sysfs: use the wrapper directly to check if the kdamond is running
Date: Mon, 12 Sep 2022 23:11:53 +0800

We can use the 'damon_sysfs_kdamond_running()' wrapper directly to check
if the kdamond is running in 'damon_sysfs_turn_damon_on()'.

Link: https://lkml.kernel.org/r/1662995513-24489-1-git-send-email-kaixuxia@tencent.com
Signed-off-by: Kaixu Xia <kaixuxia@tencent.com>
Reviewed-by: SeongJae Park <sj@kernel.org>
Reviewed-by: Muchun Song <songmuchun@bytedance.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/sysfs.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/damon/sysfs.c~mm-damon-sysfs-use-the-wrapper-directly-to-check-if-the-kdamond-is-running
+++ a/mm/damon/sysfs.c
@@ -2465,8 +2465,7 @@ static int damon_sysfs_turn_damon_on(str
 	struct damon_ctx *ctx;
 	int err;
 
-	if (kdamond->damon_ctx &&
-			damon_sysfs_ctx_running(kdamond->damon_ctx))
+	if (damon_sysfs_kdamond_running(kdamond))
 		return -EBUSY;
 	if (damon_sysfs_cmd_request.kdamond == kdamond)
 		return -EBUSY;
_
